Packager refactor

joshuaclayton opened the pull request.

qrush commented on lib/kumade/packagers/jammit_packager.rb:
```diff
@@ -0,0 +1,20 @@
+    end
+
+    def self.installed?
+      !!defined?(Jammit)
```
gross, why `!!` ?

joshuaclayton replied:
i want it to return a boolean.

``` ruby
>> defined?(FuckYeahNickQuaranto)
=> nil
>> defined?(FuckYeahJoshClayton)
=> "constant"
```

qrush replied:
hurray, a core ruby ? method that returns non booleans

![](http://jasonlefkowitz.net/wp-content/uploads/2011/08/facepalm_picard21.jpg)

joshuaclayton replied:
yeah... believe me, i tried to avoid it.

qrush commented on lib/kumade/packagers/jammit_packager.rb:
```diff
@@ -0,0 +1,20 @@
+module Kumade
+  class JammitPackager
+    def self.assets_path
+      File.join(Jammit::PUBLIC_ROOT, Jammit.package_path)
```
Any reason why `PUBLIC_ROOT` isn't a `Pathname` ?

joshuaclayton replied:
oh, does Jammit do that? that'd be pretty nice.

qrush replied:
I can't read, we should just leave it...jammit probably doesnt have one

Commit:
Refactor Kumade::Packager

This moves the logic of each of the packagers into their own classes,
introduces a PackagerList class that maintains all the various
packagers, and improves tests around the packager.
